### PR TITLE
Hubert/ Wrong error message when selecting VRTC account on payment agent page.

### PR DESCRIPTION
**Pull request by hubert-deriv**

### CODE REVIEW

File: src/javascript/app/pages/cashier/payment_agent_list.js [hunk 1]
```diff
@@ -94,8 +94,16 @@ const PaymentAgentList = (() => {
...
-        $('#no_paymentagent').setVisibility(1);
+        const currency = Client.get('landing_company_shortcode');
+        if (currency === 'virtual') {
+            $('.barspinner').setVisibility(0);
+            $('#virtual_currency').setVisibility(1);
+            $('#no_paymentagent').setVisibility(0);
+        } else {
+            $('.barspinner').setVisibility(0);
+            $('#virtual_currency').setVisibility(0);
+            $('#no_paymentagent').setVisibility(1);
```
Comment (amirghadimi-fs) [inline src/javascript/app/pages/cashier/payment_agent_list.js:105]: Please use our getElementById function to select a dom by id.
```const getElementById     = require('./common_functions').getElementById;```

Please check if we can handle this using `data-show`

Comment (hubert-deriv) [reply src/javascript/app/pages/cashier/payment_agent_list.js]: I could indeed make use of data-show handles. It should be fixed now.

File: src/javascript/app/pages/cashier/payment_agent_list.js [hunk 2]
```diff
@@ -95,7 +96,7 @@ const PaymentAgentList = (() => {
 
     const showEmptyListMsg = () => {
         $('.barspinner').setVisibility(0);
-        $('#no_paymentagent').setVisibility(1);
+        getElementById('no_paymentagent').setVisibility(1);
```
Comment (negarn) [inline src/javascript/app/pages/cashier/payment_agent_list.js:99]: why? this file is using jquery

File: src/templates/app/cashier/payment_agent_list.jsx [hunk 3]
```diff
@@ -58,7 +58,8 @@ const PaymentAgentList = () => (
             </div>
         </div>
 
-        <p id='no_paymentagent' className='notice-msg center-text invisible'>{it.L('Payment Agent services are not available in your country or in your preferred currency.')}</p>
+        <p data-show='-virtual' id='no_paymentagent' className='notice-msg center-text invisible'>{it.L('Payment Agent services are not available in your country or in your preferred currency.')}</p>
+        <p data-show='virtual' id='virtual_currency' className='notice-msg center-text'>{it.L('Virtual accounts do not have Payment Agent services.')}</p>
```
Comment (negarn) [inline src/templates/app/cashier/payment_agent_list.jsx:62]: do we need this id `virtual_currency`?

File: src/javascript/app/pages/cashier/payment_agent_list.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
-const BinaryPjax   = require('../../base/binary_pjax');
-const Client       = require('../../base/client');
-const BinarySocket = require('../../base/socket');
-const Url          = require('../../../_common/url');
+const BinaryPjax        = require('../../base/binary_pjax');
+const Client            = require('../../base/client');
+const BinarySocket      = require('../../base/socket');
+const Url               = require('../../../_common/url');
```
Comment (negarn) [inline src/javascript/app/pages/cashier/payment_agent_list.js:4]: why all the extra space?

Comment (hubert-deriv) [reply src/javascript/app/pages/cashier/payment_agent_list.js]: Resolved the issue.

**Review by negarn — APPROVED** [review]

**Pull request merged**